src/schema/document/default_document.rs

```diff
@@ -150,7 +151,7 @@ impl TantivyDocument {
     }
 
     /// Add a dynamic object field
-    pub fn add_object(&mut self, field: Field, object: BTreeMap<String, OwnedValue>) {
+    pub fn add_object(&mut self, field: Field, FlatObject(object): FlatObject) {
```

Comment:
that seems like a worse API

Reply:
Please expand on worse in what way or rather what you would prefer?

It definitely adds a bit of ceremony for the wrapper type, but it does avoid the unnecessary conversion for example in the unit tests here. Would you prefer some kind of implicit conversion via generics and trait bounds?

Reply:
`BTreeMap` is a well know data structure for objects, while `FlatObject` would require some digging on how to convert into it